### PR TITLE
HTS-1076: Unified connectors talking to DES into one

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala [hunk 1]
```diff
@@ -16,8 +16,71 @@
...
 
-import uk.gov.hmrc.http.HttpResponse
+import cats.Show
+import cats.syntax.show._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{JsNull, JsValue, Writes}
+import uk.gov.hmrc.helptosave.config.{AppConfig, WSHttp}
+import uk.gov.hmrc.helptosave.metrics.Metrics
+import uk.gov.hmrc.helptosave.models.UCResponse
+import uk.gov.hmrc.helptosave.util.Time.nanosToPrettyString
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: unused import

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala [hunk 2]
```diff
@@ -16,8 +16,71 @@
...
+@Singleton
+class DESConnectorImpl @Inject() (http:              WSHttp,
+                                  metrics:           Metrics,
+                                  pagerDutyAlerting: PagerDutyAlerting)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
+  extends DESConnector with Logging {
+
+  val itmpBaseURL: String = appConfig.baseUrl("itmp-eligibility-check")
+  val itmpEnrolmentURL: String = appConfig.baseUrl("itmp-enrolment")
+  val payeURL: String = appConfig.baseUrl("paye-personal-details")
+  val itmpThresholdURL: String = s"${appConfig.baseUrl("itmp-threshold")}/universal-credits/threshold-amount"
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: all these base urls should be pointing to  the the same DES URL , so you can cleanup this

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala [hunk 3]
```diff
@@ -16,8 +16,71 @@
...
+
+  def getPersonalDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def getThreshold()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+}
+
+@Singleton
+class DESConnectorImpl @Inject() (http:              WSHttp,
+                                  metrics:           Metrics,
+                                  pagerDutyAlerting: PagerDutyAlerting)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: why do we still have metrics and pagerduty injected here ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: same in HelpToSaveProxyConnector too

File: app/uk/gov/hmrc/helptosave/controllers/EnrolmentBehaviour.scala [hunk 4]
```diff
@@ -18,25 +18,34 @@ package uk.gov.hmrc.helptosave.controllers
...
+import play.api.http.Status
 
 import scala.concurrent.{ExecutionContext, Future}
 
 trait EnrolmentBehaviour {
 
   val enrolmentStore: EnrolmentStore
 
-  val itmpConnector: ITMPEnrolmentConnector
+  val desConnector: DESConnector
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/controllers/EnrolmentBehaviour.scala]: why connector here . ? shouldnt it be `helptosaveservice` ?

File: app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala [hunk 5]
```diff
@@ -23,23 +23,23 @@ import play.api.libs.json.{JsError, JsSuccess}
...
-                                      proxyConnector:              HelpToSaveProxyConnector,
-                                      userCapService:              UserCapService,
-                                      val eligibilityCheckService: EligibilityCheckService,
-                                      override val authConnector:  AuthConnector)(
+class HelpToSaveController @Inject() (val enrolmentStore:         EnrolmentStore,
+                                      val desConnector:           DESConnector,
+                                      proxyConnector:             HelpToSaveProxyConnector,
+                                      userCapService:             UserCapService,
+                                      val helpToSaveService:      HelpToSaveService,
+                                      override val authConnector: AuthConnector)(
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: are we using desConnector and  helpToSaveService in this class ?

File: app/uk/gov/hmrc/helptosave/controllers/StrideController.scala [hunk 6]
```diff
@@ -24,16 +24,16 @@ import play.api.libs.json.Json
...
+import uk.gov.hmrc.helptosave.services.HelpToSaveService
 import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, WithMdcExecutionContext}
 import uk.gov.hmrc.helptosave.util.Logging._
 
-class StrideController @Inject() (val eligibilityCheckService:  EligibilityCheckService,
-                                  payePersonalDetailsConnector: PayePersonalDetailsConnector,
-                                  authConnector:                AuthConnector,
-                                  enrolmentStore:               EnrolmentStore)(implicit transformer: LogMessageTransformer, override val appConfig: AppConfig)
+class StrideController @Inject() (val helpToSaveService: HelpToSaveService,
+                                  dESConnector:          DESConnector,
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/controllers/StrideController.scala]: same as above

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+import java.util.UUID
+
+import cats.data.EitherT
+import cats.syntax.eq._
+import cats.syntax.either._
+import cats.instances.int._
+import cats.instances.future._
+import configs.syntax._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala]: unused imports

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: removed, thanks

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: good call, thanks, this has now been cleaned up

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala (hunk 3 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: we don't, sorry I overlooked these in the DesConnector. By removing them from the HelpToSaveProxyConnector, the metrics and pd logic would need to be moved out, I think that is beyond the scope of this ticket?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: ok, i am not sure why the ticket says inly about DES connectors, but i meant its about all connectors logic moving to services... better if we remove them from HelpToSaveProxyConnector too and move to services..

File: app/uk/gov/hmrc/helptosave/controllers/EnrolmentBehaviour.scala (hunk 4 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/controllers/EnrolmentBehaviour.scala]: yes you're right

File: app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala (hunk 5 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: the desConnector isnt being used so I have removed it. The helpToSaveService is being used in the EnrolmentBehaviour trait so its needed here.

File: app/uk/gov/hmrc/helptosave/controllers/StrideController.scala (hunk 6 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/controllers/StrideController.scala]: I have removed the desConnector here as its not used, thanks.

File: app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala (hunk 7 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala]: removed, thanks

File: app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala [hunk 8]
```diff
@@ -42,21 +42,18 @@ trait DESConnector {
...
+class DESConnectorImpl @Inject() (http: WSHttp)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
   extends DESConnector with Logging {
 
-  val itmpBaseURL: String = appConfig.baseUrl("itmp-eligibility-check")
-  val itmpEnrolmentURL: String = appConfig.baseUrl("itmp-enrolment")
-  val payeURL: String = appConfig.baseUrl("paye-personal-details")
-  val itmpThresholdURL: String = s"${appConfig.baseUrl("itmp-threshold")}/universal-credits/threshold-amount"
+  val itmpBaseURL: String = appConfig.baseUrl("itmp")
+
+  val itmpThresholdURL: String = s"${appConfig.baseUrl("itmp")}/universal-credits/threshold-amount"
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala]: `appConfig.baseUrl("itmp")` -> itmpBaseURL

**Comment by sureshhmrc** [comment]
are all tests passing ?

**Comment by Jackie1988** [comment]
Locally all tests are passing

**Pull request merged**